### PR TITLE
Add support for pulling Spin OCI images

### PR DESCRIPTION
**Pull request by jsturtevant**
Replaces #164

### CODE REVIEW

File: containerd-shim-spin/src/engine.rs [hunk 1]
```diff
@@ -38,21 +42,102 @@ impl TriggerHooks for StdioTriggerHook {
...
+        let resolve_app_source = match app_source {
+            AppSource::File(source) => ResolvedAppSource::File {
+                manifest_path: source.clone(),
+                manifest: spin_manifest::manifest_from_file(source.clone())?,
+            },
+            AppSource::Oci => {
+                let working_dir = PathBuf::from("/");
+                let loader = spin_oci::OciLoader::new(working_dir);
+
+                // TODO: what is the best way to get this info? It isn't used only saved in the locked file
```
Comment (Mossaka) [inline containerd-shim-spin/src/engine.rs:117]: Why is this needed?

**Review by Mossaka — APPROVED** [review]
LGTM, thanks!

**Review by radu-matei — APPROVED** [review]
Tested this in a new k3d cluster with a Spin app published to a registry using `spin registry push` and everything works great!

Thank you so much for the work here!

File: containerd-shim-spin/src/engine.rs (hunk 1 above)
Comment (jsturtevant) [reply containerd-shim-spin/src/engine.rs:117]: it is part of the API to load OCI.  https://github.com/fermyon/spin/blob/40bbead627ef6c8f61706516e5f11f1122a33f21/crates/oci/src/loader.rs#L53-L58

The vaule is stored as metadata but not used anywhere.  We could probably pass this info along to the shim somehow or maybe make it optional in spin?

**Comment by Mossaka** [comment]
This can be a follow-up task. Could you please add at least one integration test for testing spin OCI images?

**Review by Mossaka — APPROVED** [review]
